### DISCUSSION TITLE
next/image return 404 when using image.path in next.config.js

### DISCUSSION THREAD

**Discussion by devinekadeni** [vercel/next.js#37609] [Help] [answered]
Hi I'm facing issue like in the screenshot when accessing static image using `next/image`
e,g:
```jsx
import Image from 'next/image'
import EmptyImage from '../../assets/img/empty-image.webp'

...
<Image src={EmptyImage} width={200} height={200} />
...
```
<img width="759" alt="Screen Shot 2022-06-10 at 15 30 14" src="https://user-images.githubusercontent.com/29223530/173024986-af48e569-865b-49f7-bd2a-938436e8c385.png">

for the context, I have configured several places related to my nextjs project such:
- using i18n (locales)
- implement `_middleware.js`
- implement `pageExtensions` in `next.config.js`
- also I create our own base-path-like mechanism in the `_middleware.js` and `next.config.js` (using rewrites)
- modify `images.path` on `next.config.js`

this is my `next.config.js` looks like
```jsx
const BASE_URL = 'http://localhost:3000'
const BASE_PREFIX_FOR_APP = 'flight'

const REWRITE_CONFIG = {
  BASE_PREFIX: {
    source: `/${BASE_PREFIX_FOR_APP}/_next/:path*`,
    destination: BASE_URL + '/_next/:path*',
  },
  PUBLIC_ASSETS: {
    source: `/${BASE_PREFIX_FOR_APP}/_next/assets/:path*`,
    destination: '/assets/:path*',
  },
};

const nextConfig = {
  reactStrictMode: true,
  pageExtensions: ['page.jsx', 'page.js'],
  i18n: {
    locales: ['en', 'id'],
    defaultLocale: 'id',
    localeDetection: false, // using middleware to determine locale
  },
  assetPrefix: '/' + BASE_PREFIX_FOR_APP, // [1]
  images: {
    // TODO: comment this to make it accessible (not 404)
    path: `/${BASE_PREFIX_FOR_APP}/_next/image`, // [1]
  },
  async rewrites() {
    return [
      REWRITE_CONFIG.BASE_PREFIX,
      REWRITE_CONFIG.PUBLIC_ASSETS,
    ];
  },
  generateBuildId: () => BASE_PREFIX_FOR_APP + '__' + nextBuildId.sync({ dir: __dirname }), // [2]
  [PHASE_DEVELOPMENT_SERVER]: {
    async rewrites() {
      return [
        REWRITE_CONFIG.BASE_PREFIX,
        REWRITE_CONFIG.PUBLIC_ASSETS,
      ];
    },
  },
}
```
After several digging, the main culprit came from `images.path`, if I comment that out it will work perfectly.
But the problem is, I need that `images.path` configuration to have that kind of prefix.

I also create reproduction github repo, please check it here https://github.com/devinekadeni/reproduce-next-image
step:
- node version 16
- `git clone`
- `yarn install`
- `yarn dev`
- go to `http://localhost:3000/flight/product`

Is there any idea how to solve this ya? Any help would be appreciated 🙏

**Comment by icyJoseph** [comment, marked as answer]
In your middleware do this change:

```ts
import { NextResponse } from "next/server";

const isMobile = true;

const getDeviceType = () => {
  return isMobile ? "mobile" : "desktop";
};

const getLocale = () => {
  const someLogic = true;

  return someLogic ? "en" : "id";
};

const PUBLIC_FILE_PATTERN = /\.(.*)$/;

export async function middleware(req) {
  const { pathname } = req.nextUrl;

  if (
    pathname.startsWith("/flight/_next/image") || // exclude all image requests
    pathname.startsWith("/api") || //  exclude all API routes - Perhaps not wanted?
    pathname.startsWith("/static") || // exclude static files - Perhaps not wanted?
    PUBLIC_FILE_PATTERN.test(pathname) // exclude all files in the public folder
  )
    return NextResponse.next();

  const locale = getLocale();
  const device = getDeviceType();
  const targetPathname = pathname.slice(1);

  const nextPageTarget = `/${locale}/${targetPathname}/${device}`;

  // mutate the target pathname
  req.nextUrl.pathname = nextPageTarget;

  const nextResponse = NextResponse.rewrite(req.nextUrl);

  return nextResponse;
}
```

The problem is that the next/image source paths are just `/flight/_next/image` and do not contain the file extension in the pathname, but as a search param named `url`, that is, `req.nextUrl.searchParams.get('url')`. IIRC, these don't normally come through the middleware, but the change in path is probably causing them to show up here.

You just gotta let them pass through. I also added the common check for API routes and static paths, but perhaps you don't want those.

Maybe this uncovers more issues though 😅

**Reply by devinekadeni** [reply]
I see... so the `_next/image` is pass through the if condition (on my example code), thank you so much for finding this out with complete explanation! 🔥 

> I also added the common check for API routes and static paths, but perhaps you don't want those. 

yes you are correct, I already have this condition too on my original project, just removing it for step to reproduce purpose 😅 

One more question tho, with your solution, is it safe if I use 
```js
pathname.include("_next")

// instead of

pathname.startsWith("/flight/_next/image")
```
I tried it and it seems safe (?)

**Comment by alaamagdyomar** [comment]
i have this issue and my project has this config
const path = require('path')

/** @type {import('next').NextConfig} */

// Remove this if you're not using Fullcalendar features
const withTM = require('next-transpile-modules')([
  '@fullcalendar/common',
  '@fullcalendar/react',
  '@fullcalendar/daygrid',
  '@fullcalendar/list',
  '@fullcalendar/timegrid'
])

module.exports = withTM({
  trailingSlash: true,
  reactStrictMode: false,
  swcMinify: false ,// it should be false by default 
  experimental: {
    esmExternals: false,
    jsconfigPaths: true // enables it for both jsconfig.json and tsconfig.json
  },
  webpack: config => {
    config.resolve.alias = {
      ...config.resolve.alias,
      apexcharts: path.resolve(__dirname, './node_modules/apexcharts-clevision')
    }

    return config
  }
})

**Comment by qwikag** [comment]
This issue has raised its head since a recent release
It was fine until I updated next.js to version 13.4.19
I am using Amplify
